api/controllers/SocketController.js

```diff
@@ -15,8 +17,15 @@ function test(message) {
 	this.socket.emit('action', { type: 'SOCKET_TEST', message: 'Hello from server!' });
 }
 
-function addTicket(ticket) {
-	this.socket.emit('action', { type: 'SOCKET_NEW_TICKET', ticket });
+async function addTicket(ticket) {
+	const { requestorId, description, tableNum, contactInfo } = ticket;
+	try{
+		const newTicket = await Ticket.create(requestorId, description, tableNum, contactInfo);
+		this.socket.emit('action', { type: 'SOCKET_NEW_TICKET', newTicket: newTicket });
+	}
+	catch(err) {
+		done(err);
```

Comment:
`done` is not a defined method here. You should `console.err()` some error I guess.

Reply:
Or just remove the `try`-`catch` altogether and allow the error to bubble up to the caller, and handle errors properly where `addTicket` is called.

Reply:
It will usually just console.err error messages. Yeah you can make an action that accepts like an error message passed in from the server.